### ISSUE TITLE
Can't use list with a single value as midinote

### ISSUE THREAD

**Issue by TatriX** [defaultxr/cl-patterns#19]
Hi!
The following patterns 
```lisp
(pb :pow
    :instrument :saw
    :quant 4
    :dur 1/4
    :midinote (list 60))
```
gives an error:
```
The value
  (60)
is not of type
  NUMBER
when binding SB-KERNEL::X
   [Condition of type TYPE-ERROR]

Restarts:
 0: [SKIP-EVENT] Skip this event, preserving the task on the clock so it can be run again.
 1: [REMOVE-TASK] Remove this task from the clock.
 2: [ABORT] abort thread (#<THREAD "cl-patterns clock-loop" RUNNING {100238F493}>)

Backtrace:
 0: (SB-KERNEL:TWO-ARG-- (60) 69.0d0) [external]
 1: (MIDINOTE-FREQ (60))
 2: (EVENT-VALUE (EVENT :PDEF :POW :INSTRUMENT :SAW :QUANT (4) :DUR 1/4 :ATK 0.01 :DEC 0.5 :SUS 0.1 :REL 0.1 :AMP 0.2 :MIDINOTE (60) :BEAT-AT-START 81/10 :TIMESTAMP-AT-START @2021-03-26T20:45:45.570964+01..
 3: ((:METHOD BACKEND-INSTRUMENT-ARGS-LIST (T T T)) :SAW (EVENT :PDEF :POW :INSTRUMENT :SAW :QUANT (4) :DUR 1/4 :ATK 0.01 :DEC 0.5 :SUS 0.1 :REL 0.1 :AMP 0.2 :MIDINOTE (60) :BEAT-AT-START 81/10 :TIMESTAMP..
 4: ((:METHOD BACKEND-PLAY-EVENT (T T T)) (EVENT :PDEF :POW :INSTRUMENT :SAW :QUANT (4) :DUR 1/4 :ATK 0.01 :DEC 0.5 :SUS 0.1 :REL 0.1 :AMP 0.2 :MIDINOTE (60) :BEAT-AT-START 81/10 :TIMESTAMP-AT-START @2021..
 5: ((:METHOD CLOCK-PROCESS-EVENT (T T T T)) #<unused argument> #<TASK :ITEM #<PDEF-PSTREAM :POW>> (EVENT :PDEF :POW :INSTRUMENT :SAW :QUANT (4) :DUR 1/4 :ATK 0.01 :DEC 0.5 :SUS 0.1 :REL 0.1 :AMP 0.2 :MID..
 6: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN CLOCK-PROCESS))
 7: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
 8: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<FUNCTION (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN CLOCK-PROCESS) {7F7E64F9664B}> #<SB-THREAD:MUTEX "Anonymous recursive lock" owner: #<SB-THREAD:THREAD "cl..
 9: (CLOCK-PROCESS #<CLOCK :tempo 1 :beat 8.0> 1/10)
10: (CLOCK-LOOP #<CLOCK :tempo 1 :beat 8.0> :GRANULARITY NIL)
11: ((LAMBDA NIL :IN START-CLOCK-LOOP))
12: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
13: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
14: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::RUN))
15: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
16: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::RUN))
17: (SB-THREAD::RUN)
18: ("foreign function: call_into_lisp")
19: ("foreign function: funcall1")

```

**Comment by defaultxr** [comment]
Looks like another bug related to multichannel expansion. :( I think the whole way expansion is done currently needs a rethink. I'll probably work on that soon. Once that's done then it should be a lot easier to implement `prest` as well.

**Comment by defaultxr** [comment]
...Actually, I think 27812ffab69ed135d1a4d537b7054882d1d8dc96 was enough to fix this. Are you still getting the error after that commit?

**Comment by TatriX** [comment]
Hm, it doesn't give an error anymore, but somehow it sends midi messages at the same time.

**Comment by TatriX** [comment]
Pattern from the first post somehow gets to `:alsa-midi` backend:
```
IN
   [Condition of type SIMPLE-CONDITION]

Restarts:
 0: [CONTINUE] Return from BREAK.
 1: [SKIP-EVENT] Skip this event, preserving the task on the clock so it can be run again.
 2: [REMOVE-TASK] Remove this task from the clock.
 3: [ABORT] abort thread (#<THREAD "cl-patterns clock-loop" RUNNING {1001887F13}>)

Backtrace:
 0: ((:METHOD BACKEND-PLAY-EVENT (T T (EQL :ALSA-MIDI))) (EVENT :PDEF :POW :INSTRUMENT :SAW :QUANT 4 :DUR 1/4 :MIDINOTE 60 :BEAT-AT-START 7481/10 :TIMESTAMP-AT-START @2021-03-28T14:47:58.425911+02:00) #<T..
 1: ((:METHOD CLOCK-PROCESS-EVENT (T T T T)) #<unused argument> #<TASK :ITEM #<PDEF-PSTREAM :POW>> (EVENT :PDEF :POW :INSTRUMENT :SAW :QUANT 4 :DUR 1/4 :MIDINOTE 60 :BEAT-AT-START 7481/10 :TIMESTAMP-AT-ST..
      Locals:
        EVENT = (EVENT :PDEF :POW :INSTRUMENT :SAW :QUANT 4 :DUR 1/4 :MIDINOTE 60 :BEAT-AT-START 7481/10 :TIMESTAMP-AT-START @2021-03-28T14:47:58.425911+02:00)
        TASK = #<TASK :ITEM #<PDEF-PSTREAM :POW>>
 2: ((FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN CLOCK-PROCESS))
 3: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::CALL-WITH-RECURSIVE-LOCK))
 4: (SB-THREAD::CALL-WITH-RECURSIVE-LOCK #<FUNCTION (FLET SB-THREAD::WITH-RECURSIVE-LOCK-THUNK :IN CLOCK-PROCESS) {7FDF530BE64B}> #<SB-THREAD:MUTEX "Anonymous recursive lock" owner: #<SB-THREAD:THREAD "cl..
 5: (CLOCK-PROCESS #<CLOCK :tempo 1 :beat 748.0> 1/10)
 6: (CLOCK-LOOP #<CLOCK :tempo 1 :beat 748.0> :GRANULARITY NIL)
 7: ((LAMBDA NIL :IN START-CLOCK-LOOP))
 8: ((LAMBDA NIL :IN BORDEAUX-THREADS::BINDING-DEFAULT-SPECIALS))
 9: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
10: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::RUN))
11: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
12: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::RUN))
13: (SB-THREAD::RUN)
14: ("foreign function: call_into_lisp")
15: ("foreign function: funcall1")
 --more--
```

**Comment by defaultxr** [comment]
OK, that should be fixed in 6c184a4e5250967d0bf8cfe2a0a9e5d781227506 now. :)

**Comment by TatriX** [comment]
Works! Thank you :)
